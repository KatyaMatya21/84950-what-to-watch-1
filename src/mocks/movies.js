export default [
  {
    'title': `Without Link`,
    'src': `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    'title': `Without Image`,
    'link': `movie-page.html`
  },
  {
    'title': `Fantastic Beasts: The Crimes of Grindelwald`,
    'src': `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Bohemian Rhapsody`,
    'src': `img/bohemian-rhapsody.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Macbeth`,
    'src': `img/macbeth.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Aviator`,
    'src': `img/aviator.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `We need to talk about Kevin`,
    'src': `img/we-need-to-talk-about-kevin.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `What We Do in the Shadows`,
    'src': `img/what-we-do-in-the-shadows.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Revenant`,
    'src': `img/revenant.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Johnny English`,
    'src': `img/johnny-english.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Shutter Island`,
    'src': `img/shutter-island.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Pulp Fiction`,
    'src': `img/pulp-fiction.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `No Country for Old Men`,
    'src': `img/no-country-for-old-men.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Snatch`,
    'src': `img/snatch.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Moonrise Kingdom`,
    'src': `img/moonrise-kingdom.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Seven Years in Tibet`,
    'src': `img/seven-years-in-tibet.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Midnight Special`,
    'src': `img/midnight-special.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `War of the Worlds`,
    'src': `img/war-of-the-worlds.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Dardjeeling Limited`,
    'src': `img/dardjeeling-limited.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Orlando`,
    'src': `img/orlando.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Mindhunter`,
    'src': `img/mindhunter.jpg`,
    'link': `movie-page.html`
  },
  {
    'title': `Midnight Special`,
    'src': `img/midnight-special.jpg`,
    'link': `movie-page.html`
  }
];
