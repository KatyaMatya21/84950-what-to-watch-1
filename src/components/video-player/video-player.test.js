import React from 'react';
import renderer from 'react-test-renderer';

import movies from "../../mocks/movies";

import VideoPlayer from './video-player.jsx';

const movie = movies[0];

describe(`VideoPlayer component`, () => {

  it(`Correctly renders`, () => {
    const VideoPlayerComponent = renderer
      .create(
          <VideoPlayer
            movie={movie}
          />,
          {
            createNodeMock: (element) => {
              if (element.type === `video`) {
                return {'current': {}};
              }
              return null;
            }
          }
      )
      .toJSON();
    expect(VideoPlayerComponent).toMatchSnapshot();
  });

});
