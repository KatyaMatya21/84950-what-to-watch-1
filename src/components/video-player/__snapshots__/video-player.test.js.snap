// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer component Correctly renders 1`] = `
<video
  height="175"
  muted={true}
  poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
  width="280"
>
  <source
    src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
</video>
`;
